src/redux/actions/samples/updateSampleFileUploadV2.js

```diff
@@ -0,0 +1,56 @@
+import moment from 'moment';
+
+import { SAMPLES_FILE_UPDATE } from 'redux/actionTypes/samples';
+import endUserMessages from 'utils/endUserMessages';
+import fetchAPI from 'utils/http/fetchAPI';
+import handleError from 'utils/http/handleError';
+
+const fileNameForApiV1 = {
+  matrix10x: 'matrix.mtx.gz',
+  barcodes10x: 'barcodes.tsv.gz',
+  features10x: 'features.tsv.gz',
+};
+
+const updateSampleFileUploadV2 = (
+  experimentId, sampleId, type, uploadStatus, uploadProgress,
+) => async (dispatch) => {
+  // Don't send an api update whenever the progress bar is updated, only for uploadStatus changes
+  // TODO: move progress to not even be a part of redux, manage it in a different way
+  if (!uploadProgress) {
```

Comment:
Instead of checking for `!uploadProgress`, is it better to check for `uploadStatus !== UploadStatus.UPLOADING` instead?